data/modules/Rondel/Rondel.lua

```diff
@@ -0,0 +1,225 @@
+-- Copyright © 2008-2021 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local Engine = require 'Engine'
+local Lang = require 'Lang'
+local Game = require 'Game'
+local Space = require 'Space'
+local cargo = require 'Commodities'
+local Comms = require 'Comms'
+local Event = require 'Event'
+local Legal = require 'Legal'
+local Serializer = require 'Serializer'
+local Equipment = require 'Equipment'
+local ShipDef = require 'ShipDef'
+local Timer = require 'Timer'
+
+--local Character = require 'Character'
+
+local l_rondel = Lang.GetResource("module-rondel")
+local l_ui_core = Lang.GetResource("ui-core")
+
+local patrol = {}
+local shipFiring = false
+local jetissionedCargo = false
+
+local rondel_victory = false -- has the player defeated all Rondel guards once in their career?
+local rondel_prize = false -- whether the player has been rewarded or not
+
+local ads = {}
+
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+	form:Clear()
+
+	form:SetTitle(l_rondel.SOLFED_INTEL)
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	form:SetMessage(l_rondel.SOLFED_INTEL_INTRO)
+
+	if option == 1 then
+		ads[ref] = nil
+		form:RemoveAdvertOnClose()
+		form:SetMessage(l_rondel.PERFECT)
+		Game.player:AddMoney(250000)
+		rondel_prize = true
+		return
+	elseif option == 2 then
+		form:SetMessage(l_rondel.YOU_VISITED_STAR_SYSTEM)
+	elseif option == 3 then
+		form:SetMessage(l_rondel.CREDITS_PROMISE)
+	end
+
+	form:AddOption(l_rondel.YOU_SHALL_HAVE_THE_DATA, 1)
+	form:AddOption(l_rondel.WHAT_IS_THIS_ABOUT, 2)
+	form:AddOption(l_rondel.NOT_INTERESTED, 3)
+
+end
+
+local onDelete = function (ref)
+	ads[ref] = nil
+end
+
+local onCreateBB = function (station)
+	if rondel_prize or not rondel_victory then return end
+
+	local ad = {
+		title    = l_rondel.HEY_OVER_HERE,
+		--message  = "",
+		station  = station,
+		--character = Character.New({armour=false}),
+	}
+
+	local ref = station:AddAdvert({
+		description = ad.title,
+		--icon        = "",
+		onChat      = onChat,
+		onDelete    = onDelete})
+	ads[ref] = ad
+end
+
+local attackShip = function (ship)
+	for i = 1, #patrol do
+		patrol[i]:AIKill(ship)
+	end
+end
+
+local onShipDestroyed = function (ship, attacker)
+	if ship:IsPlayer() or attacker == nil or Game.system.name ~= "Rondel" then return end
```

Comment:
I'd recommend comparing the SystemPath rather than the name of the system; it's entirely possible there will be multiple duplicate systems with the same name when you only meant to refer to one.